Sources/StreamChat/StateLayer/Chat.swift

```diff
@@ -660,72 +685,82 @@ public class Chat {
     ///
     /// - Throws: An error while communicating with the Stream API.
     /// - Returns: An array of messages for the pagination.
-    @discardableResult public func loadReplies(of messageId: MessageId, pagination: MessagesPagination) async throws -> [ChatMessage] {
+    @discardableResult public func loadReplies(for messageId: MessageId, pagination: MessagesPagination) async throws -> [ChatMessage] {
         let messageState = try await makeMessageState(for: messageId)
-        return try await messageUpdater.loadReplies(of: messageId, pagination: pagination, cid: cid, paginationStateHandler: messageState.replyPaginationHandler)
+        return try await messageUpdater.loadReplies(for: messageId, pagination: pagination, cid: cid, paginationStateHandler: messageState.replyPaginationHandler)
     }
 
     // MARK: -
     
-    /// Loads replies for the first page of the specified message and updates ``MessageState/replies``.
+    /// Loads more replies before the specified reply id and updates ``MessageState/replies``.
     ///
-    /// - Note: Loading the first page resets the ``MessageState/replies``.
+    /// - Parameters:
+    ///   - replyId: The message id of the reply from which older messages are loaded. If nil, the oldest currently loaded message id in ``MessageState/replies`` is used.
+    ///   - parentMessageId: The parent message id which has replies.
+    ///   - limit: The limit for the page size. The default limit is 25.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    public func loadReplies(before replyId: MessageId?, for parentMessageId: MessageId, limit: Int? = nil) async throws {
+        let messageState = try await makeMessageState(for: parentMessageId)
+        return try await messageUpdater.loadReplies(for: parentMessageId, before: replyId, limit: limit, cid: cid, paginationStateHandler: messageState.replyPaginationHandler)
+    }
+    
+    /// Loads more replies after the specified reply id and updates ``MessageState/replies``.
     ///
     /// - Parameters:
-    ///   - message: The parent message id which has replies.
+    ///   - replyId: The message id of the reply from which newer messages are loaded. If nil, the newest currently loaded message id in ``MessageState/replies`` is used.
+    ///   - parentMessageId: The parent message id which has replies.
     ///   - limit: The limit for the page size. The default limit is 25.
     ///
     /// - Throws: An error while communicating with the Stream API.
-    public func loadRepliesFirstPage(of messageId: MessageId, limit: Int? = nil) async throws {
-        let messageState = try await makeMessageState(for: messageId)
-        return try await messageUpdater.loadRepliesFirstPage(of: messageId, limit: limit, cid: cid, paginationStateHandler: messageState.replyPaginationHandler)
+    public func loadReplies(after replyId: MessageId?, for parentMessageId: MessageId, limit: Int? = nil) async throws {
```

Comment:
I think we might need to discuss if we want replies to be in the `Chat` object 🤔 Especially with Threads v2, it could make sense to create a `Thread` and `ThreadList` objects. If we have `Chat.loadReplies` and `Thread.loadReplies` will be confusing I guess 🤔

Reply:
That is a fair question. The current setup involves using `MessageState` with parent message id and reading replies from there.